ext4/internal/compactext4/compact.go

```diff
@@ -749,7 +756,7 @@ func (w *Writer) seekBlock(block uint32) {
 	if w.err != nil {
 		return
 	}
-	_, w.err = w.f.Seek(w.pos, io.SeekStart)
+	_, w.err = w.f.Seek(w.startOffset+w.pos, io.SeekStart)
```

Comment:
Can we make like an `OffsetReadWriteSeeker` type, that allows seeking but automatically adjusts for the offset? Then we don't need to make changes like this in other packages.